core/src/main/java/com/graphhopper/routing/BidirPathExtractor.java

```diff
@@ -0,0 +1,130 @@
+/*
+ *  Licensed to GraphHopper GmbH under one or more contributor
+ *  license agreements. See the NOTICE file distributed with this work for
+ *  additional information regarding copyright ownership.
+ *
+ *  GraphHopper GmbH licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except in
+ *  compliance with the License. You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.graphhopper.routing;
+
+import com.graphhopper.routing.weighting.TurnWeighting;
+import com.graphhopper.routing.weighting.Weighting;
+import com.graphhopper.storage.Graph;
+import com.graphhopper.storage.SPTEntry;
+import com.graphhopper.util.EdgeIterator;
+import com.graphhopper.util.EdgeIteratorState;
+import com.graphhopper.util.StopWatch;
+
+/**
+ * Builds a {@link Path} from the two fwd- and bwd-shortest path tree entries of a bidirectional search
+ *
+ * @author Peter Karich
+ * @author easbar
+ */
+public class BidirPathExtractor {
+    private final Graph graph;
+    private final Weighting weighting;
+    protected final Path path;
+
+    public static Path extractPath(Graph graph, Weighting weighting, SPTEntry fwdEntry, SPTEntry bwdEntry, double weight) {
+        return new BidirPathExtractor(graph, weighting).extract(fwdEntry, bwdEntry, weight);
+    }
```

Comment:
If this is the only required public method, can we make the constructor and other methods (package) protected or even private?

Reply:
the idea was that one can override stuff like 'onEdge' if need be, but `protected` works